radix-engine/src/system/system_callback.rs

```diff
@@ -113,42 +150,110 @@ pub struct System<C: SystemCallbackObject> {
+            }
+
+            if let Some(system_overrides) = &init_input.system_overrides {
+                if system_overrides.disable_auth {
+                    enabled_modules &= !EnabledModules::AUTH;
+                }
+                if system_overrides.disable_costing {
+                    enabled_modules &= !EnabledModules::COSTING;
+                }
+                if system_overrides.disable_limits {
+                    enabled_modules &= !EnabledModules::LIMITS;
+                }
+            }
+
+            enabled_modules
+        };
+
+        if let Some(system_overrides) = init_input.system_overrides {
+            if let Some(costing_override) = system_overrides.costing_parameters {
+                system_parameters.costing_parameters = costing_override;
+            }
+
+            if let Some(limits_override) = system_overrides.limit_parameters {
+                system_parameters.limit_parameters = limits_override;
+            }
+
+            if let Some(network_definition) = system_overrides.network_definition {
+                system_parameters.network_definition = network_definition;
+            }
+        }
+
+        let txn_runtime_module = TransactionRuntimeModule::new(
+            system_parameters.network_definition,
             executable.intent_hash().to_hash(),
-            executable.auth_zone_params().clone(),
-            SystemLoanFeeReserve::new(&costing_parameters, executable.costing_parameters()),
-            FeeTable::new(),
-            executable.payload_size(),
-            executable.num_of_signature_validations(),
-            execution_config,
+        );
+
+        let auth_module = AuthModule::new(executable.auth_zone_params().clone());
+        let limits_module = { LimitsModule::from_params(system_parameters.limit_parameters) };
+
+        let costing_module = CostingModule {
+            fee_reserve: SystemLoanFeeReserve::new(
+                &system_parameters.costing_parameters,
+                executable.costing_parameters(),
+            ),
+            fee_table: FeeTable::new(),
+            tx_payload_len: executable.payload_size(),
+            tx_num_of_signature_validations: executable.num_of_signature_validations(),
+            max_per_function_royalty_in_xrd: system_parameters.max_per_function_royalty_in_xrd,
+            cost_breakdown: if init_input.enable_cost_breakdown {
+                Some(Default::default())
+            } else {
+                None
+            },
+            on_apply_cost: Default::default(),
+        };
+
+        let mut modules = SystemModuleMixer::new(
+            enabled_modules,
+            KernelTraceModule,
+            txn_runtime_module,
+            auth_module,
+            limits_module,
+            costing_module,
+            ExecutionTraceModule::new(
+                init_input
+                    .execution_trace
+                    .unwrap_or(MAX_EXECUTION_TRACE_DEPTH),
```

Comment:
`unwrap_or(0)` so to disable trace?

Reply:
Trace is already disabled through the module. But can change this to zero as well.